packages/orchestration/src/utils/result-meta.js

```diff
@@ -0,0 +1,131 @@
+import { makeTracer } from '@agoric/internal';
+import { heapVowTools } from '@agoric/vow';
+
+/**
+ * @import {TraceLogger} from '@agoric/internal';
+ * @import {ERef} from '@endo/far';
+ * @import {EVow} from '@agoric/vow';
+ */
+
+const { when } = heapVowTools;
```

Comment:
please comment with justification for heap-based vow tools. What guarantees are there that the vow processing won't be lost during an upgrade.